### PR TITLE
remove unneded status checking

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Per the docs that I attached on each commit, these functions don't modify the status and so shouldn't fail. This is an effort to implement #320

### CODE REVIEW

File: src/font/font_face.rs [hunk 1]
```diff
@@ -26,16 +26,14 @@ pub struct FontFace(ptr::NonNull<ffi::cairo_font_face_t>);
 
 impl FontFace {
     pub fn toy_create(family: &str, slant: FontSlant, weight: FontWeight) -> FontFace {
```
Comment (sdroege) [inline src/font/font_face.rs:28]: What will happen here if you pass an invalid font family?

Comment (bilelmoussaoui) [reply src/font/font_face.rs:28]: Hmm I somehow completely forgot to ask myself that ^^

**Comment by bilelmoussaoui** [comment]
So I have checked the source code of cairo this time, and all of these functions might fail. In such case they should return a Result. I will do that change already :)

**Comment by sdroege** [comment]
> and all of these functions

*all*, including for example `set_antialias()`?!

**Comment by bilelmoussaoui** [comment]
> `set_antialias`

It seems https://github.com/freedesktop/cairo/blob/master/src/cairo.c#L1104-L1111

**Comment by sdroege** [comment]
Right, so if we have everything return a `Result` then the cairo API will be really hard to use...

Can it fail in any reasonable circumstances? If not then this should stay a panic

**Comment by bilelmoussaoui** [comment]
aha Indeed, panic is what I completely missed. I'm closing this one and I have just made the `Result<T, Status>` to `Result<T, Error>` switch

**Pull request closed without merging**